app/Repositories/SeatRepository.php

```diff
@@ -35,4 +38,77 @@ public function deleteSeat($id)
             return $e;
         }
     }
+
+    public function getAvailableSeats($schedule, $shift = null)
+    {
+        return $this->model->whereNotIn('id', function ($query) use ($schedule) {
+            $query->from(DB::raw('work_schedules, schedule_seat'))
+                ->select('seat_id')
+                ->where(DB::raw('`work_schedules`.`id`'), DB::raw('schedule_seat.work_schedule_id'))
+                ->where('date', Carbon::parse($schedule->date)->format('Y-m-d'))
+                ->where(function ($query) use ($schedule) {
+                    $query->where('work_schedules.shift', config('site.shift.all'));
+                    if (func_num_args() === 2) {
+                        $query->orWhere('work_schedules.shift', $shift);
+                    } else {
+                        $query->orWhere('work_schedules.shift', $schedule->shift);
+                    }
+                })
+                ->where(function ($query) {
+                    if (!empty($schedule->seat_id)) {
+                        $query->where('seat_id', $schedule->seat_id);
+                    }
+                })
+                ->whereNotNull('seat_id')
+                ->where('user_id', '!=', $schedule->user_id)
+                ->get();
+        })
+        ->get()
+        ->pluck('name', 'id');
+    }
+
+    public function getSaveData($requestData, $shift)
+    {
+        $saveData = [];
+        if (isset($requestData[$shift])) {
+            $data = $requestData[$shift];
+            foreach ($data as $key => $value) {
+                $saveData = array_merge($saveData, [[
+                    'work_schedule_id' => $key,
+                    'seat_id' => $value,
+                    'shift' => config('site.shift.' . $shift),
+                ]]);
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Repositories/SeatRepository.php</br>**Line**: 80</br>Comma required after last value in array declaration